src/modem/mod.rs

```diff
@@ -308,6 +349,54 @@ impl<'c, P: ModemPower> Modem<'c, P> {
         Ok(())
     }
 
+    /// Resets the network priority to the priority provided when initializing [Modem::init] with [NetworkModeConfig::Automatic]
+    ///
+    /// If not initialized with [NetworkModeConfig::Automatic], this function has no effect
+    pub fn reset_network_priority(&mut self) {
+        self.current_network_priority = self.user_network_priority.clone();
+    }
+
+    /// Connect to the first available radio access technology (RAT).
+    /// If connected using LTE-CatM or GSM, set that RAT as first priority for next registration attempt
+    async fn automatic_registration(
+        &self,
+        commands: &CommandRunnerGuard<'_>,
+    ) -> Result<SystemMode, Error> {
+        for mode in &self.current_network_priority {
+            match mode {
+                SystemMode::LteCatM1 => {
+                    commands.run(cnmp::SetNetworkMode(NetworkMode::Lte)).await?;
+                    commands.run(cmnb::SetNbMode(NbMode::CatM)).await?;
+                }
+                SystemMode::Gsm => {
+                    commands.run(cnmp::SetNetworkMode(NetworkMode::Gsm)).await?;
+                }
+                SystemMode::LteNbIot => {
+                    commands.run(cnmp::SetNetworkMode(NetworkMode::Lte)).await?;
+                    commands.run(cmnb::SetNbMode(NbMode::NbIot)).await?;
+                }
+                _ => {
+                    log::error!("Unsupported network mode: {:?}", mode);
```

Comment:
If "NoService" is an unsupported network mode, does that mean we always have service? :thinking:

Reply:
yes, we literally solved all the world's problems with this one